HIP/hip-1068.md

```diff
@@ -0,0 +1,528 @@
+
+The appropriate solution is one that enables low barrier to entry for users but also maintains network security
+regarding fee payment. To this end a flow that allows other users to sponsor the fees required of an account during a
+transaction submission would satisfy both desires. 
+
+Ultimately the current Hedera ecosystem lacks a native capability for an Owner account to cover transaction fees of a
+Spender without co-signing for every transaction. This limitation hinders the development of user-friendly applications
+and services that could benefit from a more flexible transaction fee payment flow.
+
+## Rationale
+
+Today the network supports 3 types of allowances `HbarAllowance`, `TokenAllowance` and `NftAllowance` . An account may
+submit a `CryptoApproveAllowance` transaction and specify the amount of value they are willing to allow another account
+to transfer out of their own balance. This HIP builds upon the allowance logic with the introduction of another
+allowance type `TransactionFeeAllowance`. 
+
+Accounts would follow the same pattern of `CryptoApproveAllowance` submission to designate and update an allowance.
+When a designated account submits a transaction identified as sponsored, the network would debit the sponsoring account
+balance for all HBar fees associated with the transaction, instead of the account sending the transaction. 
+
+Notably, a pseudo sponsored transaction flow exists today if the sponsor submits a transaction signed by the sender.
+However, this flow requires significant offline coordination and is time limited in some cases depending on transaction
+validation timing window. This HIP provides a solution that is not limited by either of these considerations. A sponsor
+need only know the accountId or EVM address of the account they want to sponsor. An account being sponsored need only
+know the account id or EVM address of their sponsor to submit a transaction and specify to the network that the
+transaction will be paid by the sponsor.
+
+Additionally, this flow allows for a user with an EVM address but no Hedera account to share their address and be
+sponsored in advance of their first transaction submission to the network. In this way their first transaction will
+both not be rejected by the network but will also automatically create their account for them offering a seamless
+onboarding process. 
+
+The proposed changes are made in alignment with the current implementation of allowances for similar capabilities
+(as seen in [HIP 336](https://github.com/hashgraph/hedera-improvement-proposal/blob/main/HIP/hip-336.md)).
+
+## User stories
+
+1. As an existing account on the network I want to be able to sponsor up to a certain HBAR limit the cost of future
+    transaction fees due by another account by granting an explicit allowance.
+2. As an existing account on the network I want to be able to sponsor an unlimited amount of HBARs to cover the cost
```

Comment:
This behaviour seems similar to the behaviour we have for NFT allowances, more specifically approvedForAll=true, where each new serial number of a given NFT will have been granted as an allowance to the spender.

Is the concept here the same? Since hbars are fungible there might be some applications like:

1. If a sponsor gives allowance to multiple accounts with unlimited HBAR amounts - the granted accounts might use a competative greedy approach to try to use a part or the whole balance for their personal use before another account and thus having unfair share of the sponsored balance
2. Defend the sponsor against a malicious account which performs operations in a loop, exhausting all of the balance

There might be safe use cases for this unlimited amount sponsorship, but I wanted to raise some security implications and corner cases, which can be thought of.

Reply:
Interesting points.
If by similar you mean will new hbars or FTs made available by a treasury be applicable to an existing allowance then yes, this would continue to be the case.

The cases noted are the same today I believe if an account was to provide one or multiple allowances with large enough values in comparison to their balance, so I don't believe this is a new edge case.
Solid consideration but I think it's already covered within the service capabilities.

It's the responsibility of the account to consider its awarded allowances.